offer-manager-domain/src/main/groovy/life/qbic/business/offers/create/CreateOfferDataSource.groovy

```diff
@@ -32,16 +31,18 @@ interface CreateOfferDataSource {
      */
     void store(Offer offer) throws DatabaseQueryException
 
+    @Deprecated
     /**
+     * This method will be replaced with the one provided by{@link life.qbic.business.offers.fetch.FetchOfferDataSource}
      * Fetches all versions of one offer id that are stored in the database
-     *
      * @param id The id for which all versions should be found
      * @return a list of all different version identifiers of an id must be at least of size 1
      */
     List<OfferId> fetchAllVersionsForOfferId(OfferId id)
```

Comment:
I am a little unsure about that, tbh. This method is only necessary for the UpdateOffer use case, where we inject an instance of type `CreateOfferDataSource`. Your suggestion would made refactoring in the UpdateOffer use case necessary, as we then would need to inject to datasources (CreateOfferDataSource and FetchOfferDataSource). Although this is technically fine, also from architectural point of view, i argue that you drift from the actual goal.
It is not related to what you are aiming for: to provide a use case that gets an OfferId and returns an Offer for it, based on the available datasource.
So I suggest you leave this at it is for now, to avoid unnecessary changes in the app code.